### PR TITLE
feat(da): DA v1.1

### PR DESCRIPTION
**Pull request by danielSanchezQ**
## 1. What does this PR implement?

Implemented aggregated proving for DA encoding.  This schema is a much lightweight design. Encoding times are improved greatly while maintaining verification ones.

## 2. Does the code have enough context to be clearly understood?

Yes. In fact is a much simpler process, a lot of code is removed.

## 3. Who are the specification authors and who is accountable for this PR?

@danielSanchezQ @thomaslavaur @megonen 

## 4. Is the specification accurate and complete?

~~No, this implies changing both encoding and verification specs and remains to be done.~~
Specification was reviwed and adapted to the new encoding schema.

## 5. Does the implementation introduce changes in the specification?

Yes. But it is in sync now.

## Checklist

* [x] 1. Description added.
* [x] 2. Context and links to Specification document(s) added.
* [x] 3. Main contact(s) (developers and specification authors) added
* [x] 4. Implementation and Specification are 100% in sync including changes. This is critical.
* [x] 5. Link PR to a specific milestone.

### CODE REVIEW

File: nomos-da/kzgrs/src/bdfg_proving.rs [hunk 1]
```diff
@@ -0,0 +1,229 @@
...
+/// This function will panic if:
+/// - The hasher fails to be constructed.
+/// - Any commitment fails to serialize properly.
+/// - The hash finalization process fails.
+#[must_use]
+pub fn generate_row_commitments_hash(commitments: &[Commitment]) -> Vec<u8> {
+    let mut hasher = Blake2bVar::new(32).expect("Hasher should be able to build");
+    for c in commitments {
+        let mut buffer = Cursor::new(Vec::new());
+        c.serialize_uncompressed(&mut buffer)
```
Comment (thomaslavaur) [inline nomos-da/kzgrs/src/bdfg_proving.rs:46]: Should we add a DST to the hash function for the Fiat-Shamir transform?

File: nomos-da/kzgrs/src/bdfg_proving.rs [hunk 2]
```diff
@@ -0,0 +1,229 @@
...
+    column_idx: usize,
+    column: &[Fr],
+    row_commitments: &[Commitment],
+    column_proof: &Proof,
+    domain: PolynomialEvaluationDomain,
+    global_parameters: &GlobalParameters,
+) -> bool {
+    let row_commitments_hash = generate_row_commitments_hash(row_commitments);
+    let h = Fr::from_le_bytes_mod_order(&row_commitments_hash);
+    let aggregated_elements: Fr = column
```
Comment (thomaslavaur) [inline nomos-da/kzgrs/src/bdfg_proving.rs:210]: We can gain a little bit of performance here by computing the powers of h before (to avoid computing it twice)

**Review by thomaslavaur — APPROVED** [review]
LGTM = Let's go to the Moon!!

File: nomos-da/kzgrs/src/bdfg_proving.rs (hunk 1 above)
Comment (danielSanchezQ) [reply nomos-da/kzgrs/src/bdfg_proving.rs:46]: I'm not sure how it is relate to fiat shamir actually. But it makes sense in case we upgrade protocols. Will do.

File: nomos-da/kzgrs/src/bdfg_proving.rs (hunk 2 above)
Comment (danielSanchezQ) [reply nomos-da/kzgrs/src/bdfg_proving.rs:210]: Yes, forgot to optimize that. Thanks for the remainder!

File: nomos-da/kzgrs/src/bdfg_proving.rs [hunk 3]
```diff
@@ -0,0 +1,248 @@
...
+///
+/// This function will panic if:
+/// - The hasher fails to be constructed.
+/// - Any commitment fails to serialize properly.
+/// - The hash finalization process fails.
+#[must_use]
+pub fn generate_row_commitments_hash(commitments: &[Commitment]) -> Vec<u8> {
+    let mut hasher = Blake2bVar::new(32).expect("Hasher should be able to build");
+    // add dst for hashing
+    hasher.update(b"NOMOS_DA_V1");
```
Comment (danielSanchezQ) [inline nomos-da/kzgrs/src/bdfg_proving.rs:45]: This needs to be added to specs

**Review by bacv — APPROVED** [review]

File: nomos-da/kzgrs/src/bdfg_proving.rs (hunk 1 above)
Comment (megonen) [reply nomos-da/kzgrs/src/bdfg_proving.rs:48]: Blake is used in another hashing step to convert the data to 31 bytes. Even though the hash input sizes are different in both cases, using a DST might be a good idea for edge cases (in case of 1 row commitment calculation)

File: nomos-da/network/core/src/protocols/replication/mod.rs [hunk 4]
```diff
@@ -171,6 +171,7 @@ mod test {
     }
 
     #[tokio::test]
+    #[ignore = "Create new fixtures"]
```
Comment (danielSanchezQ) [inline nomos-da/network/core/src/protocols/replication/mod.rs:174]: @CHr15F0x , can you generate the new fixtures? Do we have a place which explains how to actually do this?

File: nomos-da/kzgrs/src/bdfg_proving.rs (hunk 1 above)
Comment (danielSanchezQ) [reply nomos-da/kzgrs/src/bdfg_proving.rs:48]: > Blake is used in another hashing step to convert the data to 31 bytes. Even though the hash input sizes are different in both cases, using a DST might be a good idea for edge cases (in case of 1 row commitment calculation)

Uhm, it was used for the column hashing, but we do not do that anymore.  I added DST already. Or where you referring to somewhere else?

**Comment by danielSanchezQ** [comment]
@megonen @thomaslavaur , ping when the specs are in sync with this implementation. Otherwise this PR is stalled.

File: nomos-da/kzgrs/src/bdfg_proving.rs (hunk 1 above)
Comment (megonen) [reply nomos-da/kzgrs/src/bdfg_proving.rs:48]: Ahh right, it was being used there and no needed anymore. So in that case, Blake will only be used here. When you’re using a hash for just one purpose, there’s actually no need for a DST. What do you think? @thomaslavaur. Since it’s already implemented, we might as well leave it as is

**Comment by megonen** [comment]
> ping when the specs are in sync with this implementation. Otherwise this PR is stalled.

@danielSanchezQ The spec is almost ready. It’s aligned with the implementation. The only difference is in naming: since the new commitments are generated using random linear combinations, they’re referred to as combined polynomial/commitment/proof in the spec. That seemed more appropriate than aggregated.
Should we keep the term aggregated in the implementation, or maybe add a note in the spec saying that aggregated and combined are used interchangeably?

**Review by megonen — APPROVED** [review]

**Comment by danielSanchezQ** [comment]
> > ping when the specs are in sync with this implementation. Otherwise this PR is stalled.
> 
> @danielSanchezQ The spec is almost ready. It’s aligned with the implementation. The only difference is in naming: since the new commitments are generated using random linear combinations, they’re referred to as combined polynomial/commitment/proof in the spec. That seemed more appropriate than aggregated. Should we keep the term aggregated in the implementation, or maybe add a note in the spec saying that aggregated and combined are used interchangeably?

I'll change the nomenclature to combined. This must be reflected in any other spec as well, is that the case?

**Comment by megonen** [comment]
> I'll change the nomenclature to combined. This must be reflected in any other spec as well, is that the case?

Yeah, I will also update them 👍
